source/internal.d.ts

```diff
@@ -109,10 +137,23 @@ Returns a boolean for whether the string is uppercased.
 */
 export type IsUpperCase<T extends string> = T extends Uppercase<T> ? true : false;
 
+/**
+Returns a boolean for whether a string is whitespace.
+*/
+export type IsWhitespace<T extends string> = T extends Whitespace
+	? true
+	: T extends `${Whitespace}${infer Rest}`
+		? IsWhitespace<Rest>
+		: false;
+
 /**
 Returns a boolean for whether the string is numeric.
 */
-export type IsNumeric<T extends string> = T extends `${number}` ? true : false;
+export type IsNumeric<T extends string> = T extends `${number}`
+	? IsWhitespace<T> extends true
```

Comment:
Can you add a comment that says this is a workaround for https://github.com/microsoft/TypeScript/issues/46109#issuecomment-930307987 ?